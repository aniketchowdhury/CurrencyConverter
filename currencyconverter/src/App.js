import './App.css';
import {Typography, Select, TextField, makeStyles,FormControl,MenuItem, Button} from "@material-ui/core";
import {useState} from "react";
import {data} from "./mainData/data";
import _ from "underscore";
import { currencyConversion, calculator } from './currencyConversion';

const useStyles = makeStyles(theme=>({
  formControl: {
    margin: theme.spacing(1),
    top:"50px",
    display:"flex",
    flexDirection:"row",
    minWidth:"fit-content"
  },
  secondForm:{
    marginBottom:"76px"
  },
  row:{
    display:"flex",
    flexDirection:"row"
  },
  resultRow:{
      display:"flex",
      flexDirection:"row",
      marginTop:"50px"
  },
  fields:{
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  }
}))

function App() {
  const classes=useStyles();
  const [money1, setMoney1] = useState();
  const [currency1, setCurrency1] = useState("");
  const [currency2, setCurrency2] = useState("");
  const [result, setResult] = useState();
  let crossValue,resultNow;
  const handleChangeMoneyValue1=(event)=>{
    let regex = /[0-9]|\./;
    if(regex.test(event.target.value))
    setMoney1(event.target.value)
  }
  const handleChange1=(event)=>{
    setCurrency1(event.target.value);
  }

  const handleChange2=(event)=>{
    setCurrency2(event.target.value);
  }

  const convertButtonClick = () =>{
    crossValue=(currencyConversion(currency1,currency2,[]));
    resultNow = calculator(money1,crossValue);
    resultNow = (_.isEqual(currency2,"JPY"))?Math.floor(resultNow):resultNow.toFixed(2);
    setResult(resultNow);
  }

  return (
    <div className="App">
      <header className="App-header">
        <Typography variant="h4">Currency Converter</Typography>
      </header>
      <div className={classes.fields}>
        <FormControl variant="outlined" className={classes.formControl}>
        <TextField
        variant="outlined"
        value={money1}
        onChange={handleChangeMoneyValue1}
        />
          <Select
            value={currency1}
            onChange={handleChange1}
          >
            {data.map(key=>
        <MenuItem value={key.currency}>{key.currency}</MenuItem>
      )}
          </Select>
        </FormControl>
        <FormControl variant="outlined" className={`${classes.formControl} ${classes.secondForm}`}>
        <TextField
        variant="outlined"
        value={result}
        InputProps={{
          readOnly: true,
        }}
        />
          <Select
            value={currency2}
            onChange={handleChange2}
          >
            {data.map(key=>
        <MenuItem value={key.currency}>{key.currency}</MenuItem>
      )}
          </Select>
        </FormControl>
      <Button variant="contained" 
      color="primary" 
      style={{width:"284px"}}
      onClick={convertButtonClick}
      disabled={_.isEmpty(money1)||_.isEmpty(currency2)||_.isEmpty(currency1)}
      >Convert
      </Button>
      </div>
    </div>
  );
}

export default App;
