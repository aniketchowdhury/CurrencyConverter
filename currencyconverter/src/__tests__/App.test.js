import App from "../App";
import {render,fireEvent} from "@testing-library/react";

describe("test Suite for Currency Conversion",()=>{
    it("App snapshot", () => {
        const { asFragment } = render(<App />);
    
        expect(asFragment()).toMatchSnapshot();
      });

    it("give all inputs",()=>{
        const { getByTestId }=render(<App />)
        fireEvent.change(getByTestId("money1"), {
            target: { value: "100" }
          });
          expect(getByTestId("money1")).toBeDefined();
          expect(getByTestId("money2")).toBeDefined();
          expect(getByTestId("convert")).toBeDefined();
    })  
})
