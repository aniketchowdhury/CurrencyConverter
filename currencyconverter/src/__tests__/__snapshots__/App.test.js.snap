// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Suite for Currency Conversion App snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Currency Converter
      </h4>
    </header>
    <div
      class="makeStyles-fields-5"
    >
      <div
        class="MuiFormControl-root makeStyles-formControl-1"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="money1"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-7"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          data-testid="currency1"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-7"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root makeStyles-formControl-1 makeStyles-secondForm-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="money2"
              readonly=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-7"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          data-testid="currency2"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-7"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
        data-testid="convert"
        disabled=""
        style="width: 284px;"
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Convert
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
